app/src/main/java/com/github/sdpcoachme/messaging/InAppNotifier.kt

```diff
@@ -59,18 +60,25 @@ class InAppNotifier(val context: Context, val store: CachingStore) {
     private fun sendMessagingNotification(notificationTitle: String, notificationBody: String, chatId: String) {
 
         // The more info we receive, the more we can customize the notification's behaviour (up until the chat itself)
-        store.isLoggedIn().thenAccept { isLoggedIn ->
+        store.isLoggedIn().thenCompose { isLoggedIn ->
+            if (isLoggedIn) store.getCurrentEmail() else CompletableFuture.completedFuture("")
+        }.thenAccept { email ->
             val intent = when {
-                !isLoggedIn ->
+                email.isEmpty() ->
```

Comment:
In the case where email is empty, what happens if the chatId is empty too? In the case where email is not empty, we have that if check but for the case of the empty email, we don't have it. Maybe you could check if that check is made internally or if you have to add an if check here.

Reply:
This check is already conducten in the LoginActivity and handled accordingly